src/Formats/JSONUtils.cpp

```diff
@@ -144,6 +145,7 @@ namespace JSONUtils
             auto name = readFieldName(in);
             auto type = tryInferDataTypeForSingleJSONField(in, settings, inference_info);
             names_and_types.emplace_back(name, type);
+            skipWhitespaceIfAny(in);
```

Comment:
`SkipComma(in)` does `skipWhitespaceIfAny(in)` before and after the comma, if it is added here only for the last iteration, isn't it better to move it outside of the cycle? (also related to the similar code below)

Reply:
Yes, it's for the last iteration, but it's not possible to move it outside of the loop, because we need to skip spaces before checking '}' in loop condition. Let's just not use `skipComma`. I will fix